### PR TITLE
Public C API (validation only)

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: test/compilation/compilation_test.c [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fizzy/fizzy.h>
+
+bool dummy()
```
Comment (gumb0) [inline test/compilation/compilation_test.c:7]: `-Wmissing-declarations` complains about this

**Review by gumb0 — APPROVED** [review]

**Comment by axic** [comment]
Should add an install target to this, or in a new PR to be merged before.

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+bool fizzy_validate(const uint8_t* wasm_binary, size_t wasm_binary_size);
```
Comment (axic) [inline include/fizzy/fizzy.h:14]: What C std are we targeting? C99? EVMC supports C99+C11. We should probably also have a ci run for that.

File: test/compilation/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# EVMC: Ethereum Client-VM Connector API.
+# Copyright 2018-2020 The EVMC Authors.
+# Licensed under the Apache License, Version 2.0.
+
+# This CMake script creates multiple additional targets to test the compilation of public headers
+# with different C standards.
+
+set(STANDARDS c_std_99 c_std_11)
```
Comment (axic) [inline test/compilation/CMakeLists.txt:8]: Never mind, this does it, nice!

**Comment by chfast** [comment]
> Should add an install target to this, or in a new PR to be merged before.

Added here.

**Comment by axic** [comment]
Do we want to add a `FIZZY_VERSION` macro in this PR?

**Comment by chfast** [comment]
> Do we want to add a `FIZZY_VERSION` macro in this PR?

Add it where?

**Comment by axic** [comment]
`fizzy.h`, the public API. Or no need yet and assume the integrator knows what version they are using? It may be premature anyway.

**Comment by chfast** [comment]
So far I was successfully avoiding version macros.

File: test/unittests/capi_test.cpp [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fizzy/fizzy.h>
+#include <gtest/gtest.h>
+
+TEST(capi, fizzy_validate)
```
Comment (axic) [inline test/unittests/capi_test.cpp:8]: #533 does not use the `fizzy_` prefix here. Do we want to follow?

Comment (chfast) [reply test/unittests/capi_test.cpp]: I'm forward-following now.

**Review by gumb0 — APPROVED** [review]

**Review by axic — APPROVED** [review]

**Pull request merged**